api/opencl_architecture.asciidoc

```diff
@@ -1027,12 +1041,33 @@ The various SVM mechanisms to access host memory from the work-items
 associated with a kernel instance are <<svm-summary-table, summarized
 above>>.
 
+=== Memory Model: Memory Consistency Model for OpenCL 1.x
+
+IMPORTANT: This memory consistency model is <<unified-spec, deprecated
+by>> version 2.0.
+
+OpenCL 1.x uses a relaxed consistency memory model; i.e. the state of memory
+visible to a work-item is not guaranteed to be consistent across the collection
+of work-items at all times.
+
+Within a work-item memory has load / store consistency.
+Local memory is consistent across work-items in a single work-group at a
+work-group barrier.
+Global memory is consistent across work-items in a single work-group at a
+work-group barrier, but there are no guarantees of memory consistency between
+different work-groups executing a kernel.
```

Comment:
Hmmm, good point.  Perhaps something like the below text.  But as you say, I'll have that in it's own PR as it changes words, rather than bury it in here.

> Local memory is consistent across work-items in a single work-group at a work-group barrier if `CLK_LOCAL_MEM_FENCE` is set in the barrier memory fence flags.
> Global memory is consistent across work-items in a single work-group at a work-group barrier if `CLK_GLOBAL_MEM_FENCE` is set in the barrier memory fence flags.
> There are no guarantees of memory consistency between different work-groups executing a kernel.

Reply:
I've put this into an issue, but it will be seconds of work to put the above wording into a PR (just not extending this one).

https://github.com/KhronosGroup/OpenCL-Docs/issues/154